inc/Engine/Common/PerformanceHints/Admin/AdminBar.php

```diff
@@ -0,0 +1,121 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Common\PerformanceHints\Admin;
+
+use WP_Admin_Bar;
+use WP_Rocket\Abstract_Render;
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Engine\Admin\Settings\AdminBarMenuTrait;
+use WP_Rocket\Engine\Common\Context\ContextInterface;
+
+class AdminBar extends Abstract_Render {
+	use AdminBarMenuTrait;
+
+	/**
+	 * ATF context.
+	 *
+	 * @var ContextInterface
+	 */
+	private $atf_context;
+
+	/**
+	 * LRC context.
+	 *
+	 * @var ContextInterface
+	 */
+	private $lrc_context;
+
+
+	/**
+	 * Constructor
+	 *
+	 * @param ContextInterface $atf_context ATF context.
+	 * @param ContextInterface $lrc_context LRC context.
+	 * @param string           $template_path Template path.
+	 */
+	public function __construct( ContextInterface $atf_context, ContextInterface $lrc_context, $template_path ) {
```

Comment:
We need only to pass the pass the factories here and use their context products.
```suggestion
	public function __construct( array $factories, $template_path ) {
```